File: XamarinStore.iOS/Views/BasketButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using MonoTouch.UIKit;
using System.Threading.Tasks;
using MonoTouch.CoreGraphics;
using MonoTouch.CoreAnimation;
using System.Drawing;
using MonoTouch.Foundation;

namespace XamarinStore
{
	class BasketButton : UIControl
	{
		static Lazy<UIImage> BasketImage = new Lazy<UIImage>(() => UIImage.FromBundle("cart"));
		BadgeView badge;
		UIImageView imageView;
		public BasketButton()
		{

			imageView = new UIImageView(BasketImage.Value.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate)){
				TintColor = UIColor.White,
			};
			this.AddSubview(imageView);

			badge = new BadgeView(){

				Frame =  new System.Drawing.RectangleF (20,5,0,0)
			};
			this.AddSubview(badge);
		}
		public int ItemsCount
		{
			get { return badge.BadgeNumber; }
			set { badge.BadgeNumber = value; }
		}

		public void UpdateItemsCount(int count)
		{
			ItemsCount = count;
			var pathAnimation = CAKeyFrameAnimation.GetFromKeyPath("transform");	
			pathAnimation.CalculationMode = CAAnimation.AnimationPaced;
			pathAnimation.FillMode = CAFillMode.Forwards;
			pathAnimation.TimingFunction = CAMediaTimingFunction.FromName(CAMediaTimingFunction.EaseOut);
//			pathAnimation.RemovedOnCompletion = false;
			pathAnimation.Duration = .2;

			var transform = CATransform3D.MakeScale (2f, 2f, 1);
			pathAnimation.Values = new [] {
				NSValue.FromCATransform3D(CATransform3D.Identity),
				NSValue.FromCATransform3D(transform),
				NSValue.FromCATransform3D(CATransform3D.Identity),
			};
			badge.Layer.AddAnimation (pathAnimation, "pulse");
		}

		const float padding = 10;
		public override void LayoutSubviews()
		{
			base.LayoutSubviews();
			var bounds = this.Bounds;
			bounds.X += padding + 15;
			bounds.Y += padding;
			bounds.Width -= padding*2;
			bounds.Height -= padding*2;
			imageView.Frame = bounds;
		}
	}
}

File: F#/XamarinStore.iOS/Views/BasketButton.fs
namespace XamarinStore

open System
open System.Collections.Generic
open System.Linq
open System.Text
open MonoTouch.UIKit
open System.Threading.Tasks
open MonoTouch.CoreGraphics
open MonoTouch.CoreAnimation
open System.Drawing
open MonoTouch.Foundation

type BasketButton() as this =
    inherit UIControl()

    let padding = 10.0f
    let BasketImage = lazy ( UIImage.FromBundle("cart"))
    let badge = new BadgeView(Frame=new System.Drawing.RectangleF(20.0f, 5.0f, 0.0f, 0.0f))
    let imageView = new UIImageView(BasketImage.Value.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate), TintColor = UIColor.White)
    do this.AddSubview imageView
       this.AddSubview badge

    member this.ItemsCount
        with get () = badge.BadgeNumber
        and set value = badge.BadgeNumber <- value

    member this.UpdateItemsCount count =
        this.ItemsCount <- count
        let pathAnimation = CAKeyFrameAnimation.GetFromKeyPath("transform")
        pathAnimation.CalculationMode <- CAAnimation.AnimationPaced.ToString()
        pathAnimation.FillMode <- CAFillMode.Forwards.ToString()
        pathAnimation.TimingFunction <- CAMediaTimingFunction.FromName(CAMediaTimingFunction.EaseOut)
        pathAnimation.Duration <- 0.2

        let transform = CATransform3D.MakeScale (2.0f, 2.0f, 1.0f)
        pathAnimation.Values <- [| NSValue.FromCATransform3D(CATransform3D.Identity)
                                   NSValue.FromCATransform3D(transform)
                                   NSValue.FromCATransform3D(CATransform3D.Identity)|]
        badge.Layer.AddAnimation (pathAnimation, "pulse")

    override this.LayoutSubviews() =
        base.LayoutSubviews()
        let mutable bounds = this.Bounds
        bounds.X <- bounds.X + padding + 15.0f
        bounds.Y <- bounds.Y + padding
        bounds.Width <- bounds.Width - padding * 2.0f
        bounds.Height <- bounds.Height - padding * 2.0f
        imageView.Frame <- bounds

